index.js

```diff
@@ -96,48 +103,103 @@ class Registrar extends Emitter {
     }
   }
 
+  /**
+   * if param realm exists then returns count of users belonging to a realm,
+   * otherwise returns count of all registered users
+   * @param {String} realm - nullable realm (e.g. drachtio.org)
+   * @returns {int} count of users
+   */
   async getCountOfUsers(realm) {
+    return this.getRegisteredUsersForRealm(realm).length;
+  }
+
+  /**
+   * if realm exists then returns all users belonging to a realm,
+   * otherwise returns all registered users
+   * @param {String} aor - the address-of-record for the user
+   * @param {String} expires - epoch milliseconds when object expires
+   * @returns {Boolean} true if the registration was successfully added
+   */
+  async addToActiveUserHash(aor, expires) {
     try {
-      const users = new Set();
-      let idx = 0;
-      const pattern = makeUserPattern(realm);
-      do {
-        const res = await this.client.scan([idx, 'MATCH', pattern, 'COUNT', 100]);
-        const next = res[0];
-        const keys = res[1];
-        debug(next, keys,  `Registrar:getCountOfUsers result from scan cursor ${idx} ${realm}`);
-        keys.forEach((k) => users.add(k));
-        idx = parseInt(next);
-      } while (idx !== 0);
-      return users.size;
+      const cleanupResult = this.cleanActiveUserHash(aor);
+      const addResult = await this.client.hset('active-user', `${expires}`, aor);
```

Comment:
based [this](https://redis.io/commands/hset/), I don't see that expiry is doing what you think it is...does HSET support expiry?  Cuz it seems more like you are setting a property named "60" (e.g if that is the expires) in the hash named 'active-user' to a value of whatever the aor is.  Or am I confused?  (Probably)

Reply:
A bug I believe, it should of been 

```
const addResult = await this.client.hset('active-user', aor, `${expires}`);
```

The idea was to hold a hset of aor's as the fields and the value being the expiry date (epoch Milliseconds). The housecleaning is done outside of redis by iterating over the hset and removing any values which have expired.

**_However_** after some digging, I believe there is a better way using a sorted set and scores: https://stackoverflow.com/a/55389321 

Setting the score as the expiry we can get all expired entries by using `this.client.zrangebyscore`

I will push changes for review